torchgeo/trainers/change.py

```diff
@@ -0,0 +1,274 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for change detection."""
+
+import os
+from typing import Any, Literal
+
+import kornia.augmentation as K
+import matplotlib.pyplot as plt
+import segmentation_models_pytorch as smp
+import torch
+import torch.nn as nn
+from einops import rearrange
+from matplotlib.figure import Figure
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.classification import (
+    BinaryAccuracy,
+    BinaryF1Score,
+    BinaryJaccardIndex,
+)
+from torchvision.models._api import WeightsEnum
+
+from ..datasets import RGBBandsMissingError, unbind_samples
+from ..models import FCSiamConc, FCSiamDiff, get_weight
+from . import utils
+from .base import BaseTask
+
+
+class ChangeDetectionTask(BaseTask):
+    """Change Detection. Currently supports binary change between two timesteps.
+
+    .. versionadded: 0.8
+    """
+
+    def __init__(
+        self,
+        model: Literal['unet', 'fcsiamdiff', 'fcsiamconc'] = 'unet',
+        backbone: str = 'resnet50',
+        weights: WeightsEnum | str | bool | None = None,
+        in_channels: int = 3,
+        pos_weight: Tensor | None = None,
+        loss: Literal['bce', 'jaccard', 'focal'] = 'bce',
+        lr: float = 1e-3,
+        patience: int = 10,
+        freeze_backbone: bool = False,
+        freeze_decoder: bool = False,
+    ) -> None:
+        """Inititalize a new ChangeDetectionTask instance.
+
+        Args:
+            model: Name of the model to use.
+            backbone: Name of the `timm
+                <https://smp.readthedocs.io/en/latest/encoders_timm.html>`__ or `smp
+                <https://smp.readthedocs.io/en/latest/encoders.html>`__ backbone to use.
+            weights: Initial model weights. Either a weight enum, the string
+                representation of a weight enum, True for ImageNet weights, False or
+                None for random weights, or the path to a saved model state dict. FCN
+                model does not support pretrained weights. Pretrained ViT weight enums
+                are not supported yet.
```

Comment:
@isaaccorley is this still true? I thought SMP now supports ViT backbones. Or is this something we need to change in our weight loader?

Reply:
This is not true anymore but will need to do some testing to make sure it works out of the box in our trainers.

Reply:
Cool, let's remove this sentence then. I'll open a separate PR to remove it from other trainers.